### DISCUSSION TITLE
Products query with search filter doesn't return anything

### DISCUSSION THREAD

**Discussion by smallwire** [saleor/saleor#15410] [Q&A]
Hi,

The following query doesn't return any products for me:

```
{
  products(first: 10, channel: "eur", filter: { search: "smth which exists for sure" }) {
    edges {
      node {
        name
      }
    }
  }
}
```

If I leave the search empty I get all the products. Any ideas on what's going on?

Thank you!

**Comment by tomaszlud** [comment]
Hey,

I noticed that the `search` function works for both product `name` and `description`.
For example, consider having: `Apple Juice`, `Blue Polygon Shirt`, `Blue Hoodie` in Saleor
If we set the `search` to `blue`, we'll get these products: `Blue Polygon Shirt`, `Blue Hoodie`
If we set the `search` to `apple`, we'll get this product: `Apple Juice`
But if we set the search to `blue apple`, we'll get an empty list.

I'd try using a simpler keyword in the `search` first.
You can also try out the new API in Saleor here: https://docs.saleor.io/docs/3.x/api-usage/filtering#search-filtering

**Comment by smallwire** [comment]
Hi Tomasz,

Thank you for checking. I did try several search queries, without success. For example:

```
{
  products(first: 10, channel: "eur", filter: { search: "" }) {
    edges {
      node {
        name
      }
    }
  }
}
```

gives me:

```
{
  "data": {
    "products": {
      "edges": [
        {
          "node": {
            "name": "First palinka"
          }
        },
        {
          "node": {
            "name": "Second palinka"
          }
        },
        {
          "node": {
            "name": "Third palinka"
          }
        },
        {
          "node": {
            "name": "Toi"
          }
        }
      ]
    }
  },
  "extensions": {
    "cost": {
      "requestedQueryCost": 10,
      "maximumAvailable": 50000
    }
  }
}
```

But

```
{
  products(first: 10, channel: "eur", filter: { search: "palinka" }) {
    edges {
      node {
        name
      }
    }
  }
}
```

returns:

```
{
  "data": {
    "products": {
      "edges": []
    }
  },
  "extensions": {
    "cost": {
      "requestedQueryCost": 10,
      "maximumAvailable": 50000
    }
  }
}
```

**Reply by BenSampo** [reply]
Did you ever get to the bottom of this @smallwire? I'm running into the same issue.

**Reply by tomaszlud** [reply]
Can you check if the product is available for purchase and published? I have attached an image for reference.
![image (2)](https://github.com/user-attachments/assets/49950a6d-392d-438a-8323-98dc2e070381)

**Reply by BenSampo** [reply]
In my case yes it is. I actually found that search works, but only matches on whole words. e.g. If you have a product called "Flash Trainers" it would match for "Trainers" but not "Train". The version of Saleor I upgraded from did match on partial words.

**Reply by kadewu** [reply]
Hi @BenSampo . Are you self-hosting Saleor? If yes please make sure that you have active celery worker running.
